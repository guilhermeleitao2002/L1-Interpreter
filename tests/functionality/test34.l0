type Btree = [
  nul:unit;
  node:{ left:Btree; val:int; right:Btree }
];
let t1 = node({ left=nul(()); val=0; right=nul(())});
let t2 = node({ left=t1; val=2; right=t1});
let t3 = node({ left=t2; val=5; right=t1});
t3;;