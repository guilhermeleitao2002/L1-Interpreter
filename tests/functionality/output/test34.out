node({ val = 5, left = node({ val = 2, left = node({ val = 0, left = nul(()), right = nul(()) }), right = node({ val = 0, left = nul(()), right = nul(()) }) }), right = node({ val = 0, left = nul(()), right = nul(()) }) })